### PR TITLE
.NET Core 3.1: Refactors all projects to use latest LTS version 

### LINKED ISSUE #2150: .NET 3.1 upgrades
All the tests, benchmarks, and sample projects should be updated to use the latest [.NET 5 version](https://dotnet.microsoft.com/platform/support/policy/dotnet-core).

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

.NET core 2.1 life cycle ends this August 21, 2021. This updates all project to use the latest .NET Core 3.1

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #2150

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml [hunk 1]
```diff
@@ -281,7 +281,7 @@ FROM root]]></SqlQuery>
     </Input>
     <Output>
       <SqlQuery><![CDATA[
-SELECT VALUE {"value": -7.9228162514264338E+28} 
+SELECT VALUE {"value": -7.922816251426434E+28}
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml:284]: @sboshra and @bchong95 can you please take a look?

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj:1]: What is the difference here?

File: Microsoft.Azure.Cosmos.Samples/Tools/CTL/CosmosCTL.csproj [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/CTL/CosmosCTL.csproj:1]: Same here

**Review by ealsur — DISMISSED** [review]
nit: Some of the files seem to show edits without changes, might want to undo those

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj (hunk 2 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/CosmosBenchmark.csproj:1]: It removes a single white space at the end. VS auto applies the formatting when every you edit the file. I figured it was better to fix the formatting and to avoid the change in future PRs.

File: Microsoft.Azure.Cosmos.Samples/Tools/CTL/CosmosCTL.csproj (hunk 3 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos.Samples/Tools/CTL/CosmosCTL.csproj:1]: Please see comment above.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml [hunk 4]
```diff
@@ -292,7 +292,7 @@ FROM root]]></SqlQuery>
     </Input>
     <Output>
       <SqlQuery><![CDATA[
-SELECT VALUE {"value": 7.9228162514264338E+28}
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml:295]: How are precision related to the runtime change?

File: Microsoft.Azure.Cosmos/src/UserAgentContainer.cs [hunk 5]
```diff
@@ -38,18 +38,41 @@ internal void SetFeatures(string features)
             this.Suffix = string.Empty;
         }
 
-        private string CreateBaseUserAgentString(string features = null)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/UserAgentContainer.cs:41]: Can it be  separate PR?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestMathFunctions.xml [hunk 6]
```diff
@@ -158,7 +158,7 @@ WHERE ((root["NumericField"] > -1000000) AND (root["NumericField"] < 1000000)) ]
     <Output>
       <SqlQuery><![CDATA[
 SELECT VALUE COS(root["NumericField"]) 
-FROM root ]]></SqlQuery>
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestMathFunctions.xml:161]: Why formatting changes?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml (hunk 4 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml:295]: It seems to be done to make it more complaint with standards: https://devblogs.microsoft.com/dotnet/floating-point-parsing-and-formatting-improvements-in-net-core-3-0/

File: Microsoft.Azure.Cosmos/src/UserAgentContainer.cs (hunk 5 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/UserAgentContainer.cs:41]: I moved it to a new PR. This is required because .NET is now blocking modifying read-only fields with reflection.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml (hunk 4 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml:295]: Test's will have to be done with the backend to determine if these numbers roundtrip with the backend's json double parser.

Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestLiteralSerialization.xml:295]: Compute and most customer's are already using .NET Core 3.1. Wouldn't we have seen an issue by now if there was an issue with the backend?

**Comment by j82w** [comment]
Created a new branch because of the merge conflits: #2660
